tools/src/launcher/Main.hs

```diff
@@ -7,34 +7,35 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
 import           Control.Concurrent           (modifyMVar_)
-import           Control.Concurrent.Async     (Async, cancel, poll, waitAny,
+import           Control.Concurrent.Async     (Async, async, cancel, poll, wait, waitAny,
                                                withAsyncWithUnmask)
 import           Data.List                    (isSuffixOf)
 import qualified Data.String.QQ               as Q
+import qualified Data.Text.IO                 as T
+import qualified Data.Text.Lazy.IO            as TL
 import           Data.Version                 (showVersion)
-import qualified Filesystem.Path              as FP
-import           Filesystem.Path.CurrentOS    (encodeString)
+import           Formatting                   (format, int, stext, string, text, (%))
 import           Options.Applicative.Simple   (Mod, OptionFields, Parser, auto,
                                                execParser, footerDoc, fullDesc, header,
                                                help, helper, info, infoOption, long,
                                                metavar, option, progDesc, short,
                                                strOption)
-import           System.Directory             (getTemporaryDirectory)
+import           Pos.Util                     (directory, sleep)
+import           Prelude                      (putStrLn)
+import           System.Directory             (createDirectoryIfMissing, doesFileExist,
+                                               getTemporaryDirectory, removeFile)
 import           System.Environment           (getExecutablePath)
-import           System.FilePath              ((</>))
+import           System.Exit                  (ExitCode (..))
+import           System.FilePath              (normalise, (</>))
 import qualified System.IO                    as IO
-import           System.Process               (ProcessHandle)
+import           System.Process               (ProcessHandle, readProcessWithExitCode)
 import qualified System.Process               as Process
 import           System.Timeout               (timeout)
 import           System.Wlog                  (lcFilePrefix)
 import           Text.PrettyPrint.ANSI.Leijen (Doc)
-import           Turtle                       (ExitCode (..), FilePath, Line, Shell,
-                                               appendonly, d, echo, fork, format, fp,
-                                               mktemp, mktree, outhandle, printf, proc,
-                                               rm, s, sh, sleep, testfile, wait, (%))
-import           Universum                    hiding (FilePath)
+import           Universum                    hiding (putStrLn)
```

Comment:
> and thus pay for the automatic O(n) packing from String -> Text

There's not going to be any packing AFAIK, because of rewrite rules applying to string literals.